cloudbuild.yaml

```diff
@@ -0,0 +1,75 @@
+steps:
+  - id: 'fetch_secret'
+    name: gcr.io/cloud-builders/gcloud
+    entrypoint: 'bash'
+    args: [ '-c', "gcloud secrets versions access latest --secret=stargate-artifactory --format='get(payload.data)' | tr '_-' '/+' | base64 -d > /workspace/ci/custom-settings.xml" ]
+  - name: 'gcr.io/cloud-builders/gsutil'
+    args:
+      - '-m'
+      - 'rsync'
+      - '-r'
+      - 'gs://${_BUCKET}/cache/.m2'
+      - '/cache/.m2'
+    volumes:
+      - path: '/cache/.m2'
+        name: 'm2_cache'
+  - id: 'build'
+    name: 'maven:3-jdk-8'
+    entrypoint: mvn
+    args: [ '-B','install', '--settings', 'ci/custom-settings.xml', '--file', 'pom.xml', '-DskipITs',
+            '-Dmaven.javadoc.skip=true', '-Pdse', '-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn'
+    ]
+    volumes:
+      - path: '/cache/.m2'
+        name: 'm2_cache'
+    env:
+      - 'MAVEN_OPTS=-Dmaven.repo.local=/cache/.m2'
+  - id: 'cassandra-3.11'
+    waitFor: [ 'build' ]
+    name: 'gcr.io/stargateio/stargate-builder:v1.0.0'
+    entrypoint: 'bash'
+    args: [ 'ci/test.sh' ]
+    volumes:
+      - path: '/cache/.m2'
+        name: 'm2_cache'
+    env:
+      - 'MAVEN_OPTS=-Dmaven.repo.local=/cache/.m2'
+      - 'PERSISTENCE_BACKEND=cassandra-3.11'
+  - id: 'cassandra-4.0'
+    waitFor: [ 'build' ]
+    name: 'gcr.io/stargateio/stargate-builder:v1.0.0'
+    entrypoint: 'bash'
+    args: [ 'ci/test.sh' ]
+    volumes:
+      - path: '/cache/.m2'
+        name: 'm2_cache'
+    env:
+      - 'MAVEN_OPTS=-Dmaven.repo.local=/cache/.m2'
+      - 'PERSISTENCE_BACKEND=cassandra-4.0'
+  - id: 'dse-6.8'
+    waitFor: [ 'build' ]
+    name: 'gcr.io/stargateio/stargate-builder:v1.0.0'
+    entrypoint: 'bash'
+    args: [ 'ci/test.sh' ]
+    volumes:
+      - path: '/cache/.m2'
+        name: 'm2_cache'
+    env:
+      - 'MAVEN_OPTS=-Dmaven.repo.local=/cache/.m2'
+      - 'PERSISTENCE_BACKEND=dse-6.8'
+  - name: 'gcr.io/cloud-builders/gsutil'
+    args:
+      - '-m'
+      - 'rsync'
+      - '-r'
+      - '/cache/.m2'
```

Comment:
Should we exclude Stargate stuff?.. just to be certain no cross-build contamination happens... IIRC, the build script _installs_ Stargate artifacts.

Reply:
Oops, excluded on the download but need to add it on the upload too